pkg/receive/multitsdb.go

```diff
@@ -202,16 +203,18 @@ func (t *MultiTSDB) Sync(ctx context.Context) error {
 		}
 		wg.Add(1)
 		go func() {
-			if uploaded, err := s.Sync(ctx); err != nil {
+			up, err := s.Sync(ctx)
+			if err != nil {
 				errmtx.Lock()
 				merr.Add(errors.Wrapf(err, "upload %d", uploaded))
 				errmtx.Unlock()
 			}
+			atomic.AddInt64(&uploaded, int64(up))
```

Comment:
If uploading the block results in an error, then we still call this, is that what we want? If so, then we are simply adding "1" for every single tenant. Rather than using an atomic counter, we could simply return the length of the `tenants`. This would be simpler code and avoid using `atomic`. If we don't want to count errored blocks, then we could still do this without `sync/atomic` by returning the length of the tenants - the length of the multi-error.

Reply:
We actually:
* can  upload more than one for tenant.
* return error also when we upload one but second failed

So I think we need something like this... WDYT?

Reply:
ack I see. makes sense :)